sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt

```diff
@@ -51,53 +51,50 @@ internal class KlaviyoTest : BaseTest() {
             // Mocking an intent to return the stub push payload...
             val intent = mockk<Intent>()
             val bundle = mockk<Bundle>()
-            var gettingKey = ""
             every { intent.extras } returns bundle
             every { bundle.keySet() } returns payload.keys
-            every {
-                intent.getStringExtra(
-                    match { s ->
-                        gettingKey = s // there must be a better way to do this...
-                        true
-                    }
-                )
-            } answers { payload[gettingKey] }
-            every {
-                bundle.getString(
-                    match { s ->
-                        gettingKey = s // there must be a better way to do this...
-                        true
-                    },
-                    String()
-                )
-            } answers { payload[gettingKey] }
+            every { intent.getStringExtra(any()) } answers { call -> payload[call.invocation.args[0]] }
+            every { bundle.getString(any(), String()) } answers { call -> payload[call.invocation.args[0]] }
 
             return intent
         }
     }
 
     private val capturedProfile = slot<Profile>()
-    private val staticClock = StaticClock(TIME, ISO_TIME)
-    private val debounceTime = 5
-    private val apiClientMock: ApiClient = mockk()
+    private val apiClientMock: ApiClient = mockk<ApiClient>().apply {
+        Registry.register<ApiClient>(this)
+        every { onApiRequest(any(), any()) } returns Unit
+        every { enqueueProfile(capture(capturedProfile)) } returns Unit
+        every { enqueueEvent(any(), any()) } returns Unit
+        every { enqueuePushToken(any(), any()) } returns Unit
+    }
+
+    private val builderMock = mockk<Config.Builder>().apply {
+        every { apiKey(any()) } returns this
+        every { applicationContext(any()) } returns this
+        every { build() } returns configMock
+    }
+
+    private val mockApplication = mockk<Application>().apply {
+        every { contextMock.applicationContext } returns this
+        every { unregisterActivityLifecycleCallbacks(any()) } returns Unit
+        every { registerActivityLifecycleCallbacks(any()) } returns Unit
+    }
 
     @Before
     override fun setup() {
         super.setup()
-        Registry.register<ApiClient> { apiClientMock }
-        every { Registry.clock } returns staticClock
-        every { apiClientMock.onApiRequest(any(), any()) } returns Unit
-        every { apiClientMock.enqueueProfile(capture(capturedProfile)) } returns Unit
-        every { apiClientMock.enqueueEvent(any(), any()) } returns Unit
-        every { apiClientMock.enqueuePushToken(any(), any()) } returns Unit
-        every { configMock.debounceInterval } returns debounceTime
+        every { Registry.configBuilder } returns builderMock
         DevicePropertiesTest.mockDeviceProperties()
-        UserInfo.reset()
+        Klaviyo.initialize(
```

Comment:
Had to adjust the setup for these tests because of how I changed `UserInfo` from an Object to a registered service that needs to be instantiated.